builtin/receive-pack.c

```diff
@@ -27,6 +27,7 @@
 #include "object-store.h"
```

Comment:
The commit message is awfully quiet about that bug that it fixes (exposed via the test failures in t5509): now, `receive.denyCurrentBranch` is handled correctly even when pushing into the current branch in a repository where said branch has no commits yet.

Please do talk about it, though. We figured this one out in this PR, and there is no reason to stop this finding from making its permanent home in Git's commit history.

Reply:
This commit message still does not talk about that bug we fixed (incidentally, I might want to add). Maybe add a paragraph like this?

> Incidentally, this change fixes the bug where `receive.denyCurrentBranch = true` was ignored when pushing into a non-bare repository's unborn current branch.

Reply:
While I see this paragraph in here now, I fear that you replaced this commit's commit message with the one intended for the first commit :-( It talks about fixing `get_main_worktree()`, but this here patch is actually about fixing `receive.denyCurrentBranch` with regards to secondary worktrees...